homeassistant/components/frank_energie/__init__.py

```diff
@@ -0,0 +1,46 @@
+"""The Frank Energie component."""
+from __future__ import annotations
+
+from python_frank_energie import FrankEnergie
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_AUTH_TOKEN, CONF_COORDINATOR, CONF_REFRESH_TOKEN, DOMAIN
+from .coordinator import FrankEnergieCoordinator
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the Frank Energie component from a config entry."""
+
+    # Initialise the coordinator and save it as domain-data
+    api = FrankEnergie(
+        clientsession=async_get_clientsession(hass),
+        auth_token=entry.data.get(CONF_AUTH_TOKEN, None),
+        refresh_token=entry.data.get(CONF_REFRESH_TOKEN, None),
+    )
+    frank_coordinator = FrankEnergieCoordinator(hass, entry, api)
+
+    # Fetch initial data, so we have data when entities subscribe and set up the platform
+    await frank_coordinator.async_config_entry_first_refresh()
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {
+        CONF_COORDINATOR: frank_coordinator,
+    }
```

Comment:
```suggestion
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = frank_coordinator
```
It's the only object you're currently storing, so no real need for a separate key